protocols/migration/migration_reports_4_0_0_to_reports_3_0_0.py

```diff
@@ -0,0 +1,91 @@
+import logging
+
+from protocols import reports_4_0_0 as reports_4_0_0
+from protocols import reports_3_0_0 as reports_3_0_0
+from protocols.migration.base_migration import BaseMigration
+from protocols.migration.base_migration import MigrationError
+
+
+class MigrateReports400To300(BaseMigration):
+
+    old_model = reports_4_0_0
+    new_model = reports_3_0_0
+
+    def migrate_clinical_report_rd(self, old_instance):
+        """
+        :type old_instance: reports_4_0_0.ClinicalReportRD
+        :rtype: reports_3_0_0.ClinicalReportRD
+        """
+        new_instance = self.convert_class(self.new_model.ClinicalReportRD, old_instance)
+
+        # interpretationRequestId changed to interpretationRequestID
+        new_instance.interpretationRequestID = old_instance.interpretationRequestId
+
+        # interpretationRequestAnalysisVersion can be null in version 4
+        if hasattr(old_instance, 'interpretationRequestAnalysisVersion') and old_instance.interpretationRequestAnalysisVersion is not None:
```

Comment:
why not just `if old_instance.interpretationRequestAnalysisVersion is not None:`

Reply:
because there's no validation on whether the object has the `interpretationRequestAnalysisVersion` attribute, it came up in testing

Reply:
Isn't this true for all attributes given that this is Python?
The field `type` in the docstring should alleviate this though. Not convinced that this must be here although it does not make any bad

Reply:
@priesgo -- nope:
```
In [1]: some_string = 'hello'

In [2]: hasattr(some_string, 'some_new_attribute')
Out[2]: False
```

Reply:
what?
I mean every time you write a function that receives any `protocols.reports_x_y_z.Class` you don't know if the function was called with an instance of that class or something else, but that does not mean that we have to check every attribute. Why these ones and not others?